build-scripts/build_xccdf.py

```diff
@@ -49,11 +54,70 @@ def parse_args():
         dest="build_ovals_dir",
         help="Directory to store OVAL document for each rule.",
     )
-    parser.add_argument("--resolved-base",
-                        help="To which directory to put processed rule/group/value YAMLs.")
+    parser.add_argument(
+        "--resolved-base",
+        help="To which directory to put processed rule/group/value YAMLs."
+    )
+    parser.add_argument(
+        "--per-profile-dir",
+        help="Directory to store XCCDF, OVAL, OCIL, per profile. (off: to disable)"
+        "e.g.: ~/scap-security-guide/build/rhel7/thin_ds_component/",
+    )
     return parser.parse_args()
 
 
+def link_oval(xccdftree, checks, output_file_name, build_ovals_dir):
+    translator = ssg.id_translate.IDTranslator("ssg")
+    oval_linker = ssg.build_renumber.OVALFileLinker(
+        translator, xccdftree, checks, output_file_name
+    )
+    oval_linker.build_ovals_dir = build_ovals_dir
+    oval_linker.link()
+    oval_linker.save_linked_tree()
+    oval_linker.link_xccdf()
+
+
+def link_ocil(xccdftree, checks, output_file_name, ocil):
+    translator = ssg.id_translate.IDTranslator("ssg")
+    ocil_linker = ssg.build_renumber.OCILFileLinker(
+        translator, xccdftree, checks, output_file_name
+    )
+    ocil_linker.link(ocil)
+    ocil_linker.save_linked_tree()
+    ocil_linker.link_xccdf()
+
+
+def link_benchmark(loader, xccdftree, args, benchmark=None):
+    checks = xccdftree.findall(".//{%s}check" % ssg.constants.XCCDF12_NS)
+
+    link_oval(xccdftree, checks, args.oval, args.build_ovals_dir)
+
+    ocil = loader.export_ocil_to_xml(benchmark)
+    if ocil is not None:
+        link_ocil(xccdftree, checks, args.ocil, ocil)
+
+    ssg.xml.ElementTree.ElementTree(xccdftree).write(args.xccdf)
+
+
+def get_path(path, file_name):
+    return os.path.join(path, file_name)
+
+
+def link_benchmark_per_profile(loader, args):
+    if not os.path.exists(args.per_profile_dir):
+        os.makedirs(args.per_profile_dir)
+
+    for id_, benchmark in loader.get_benchmark_by_profile():
+        xccdftree = benchmark.to_xml_element(loader.env_yaml)
+        p = Paths_(
+            xccdf=get_path(args.per_profile_dir, "xccdf_{}.xml".format(id_)),
+            oval=get_path(args.per_profile_dir, "oval_{}.xml".format(id_)),
+            ocil=get_path(args.per_profile_dir, "ocil_{}.xml".format(id_)),
```

Comment:
Can we not include OCIL at all?

Reply:
If the rule does not contain OCIL. OCIL does not appear in the final ds.

Reply:
So I'd keep it.

Reply:
Yes, it's simple it's just a few lines of code.

Reply:
[Disable build of OCIL for thin ds](https://github.com/ComplianceAsCode/content/pull/11549/commits/241cac43e78818c25e3c735108462390e12418b4)